save-frontend/src/main/kotlin/com/saveourtool/save/frontend/components/TopBar.kt

```diff
@@ -54,6 +54,148 @@ external interface TopBarProps : PropsWithChildren {
+        findExclude(href)
+    }
+
+    /**
+     * The function is called to specify the link address in the button before creating the button itself
+     *
+     * @param pathPart is an appended suffix to an already existing [currentPath]
+     */
+    fun changeUrlBeforeButton(pathPart: String) {
+        currentPath = when (circumstance) {
+            SituationUrlClassification.PROJECT_OR_ORGANIZATION -> "#/${FrontendRoutes.PROJECTS.path}"
+            SituationUrlClassification.ARCHIVE -> "#/${FrontendRoutes.AWESOME_BENCHMARKS.path}"
+            SituationUrlClassification.DETAILS, SituationUrlClassification.EXECUTION -> if (pathPart == "execution") currentPath else mergeUrls(pathPart)
+            else -> mergeUrls(pathPart)
+        }
+    }
+
+    /**
+     * The function is called to specify the link address in the button after creating the button itself
+     *
+     * @param pathPart is an appended suffix to an already existing [currentPath]
+     */
+    fun changeUrlAfterButton(pathPart: String) {
+        fixCurrentPathAfter(pathPart)
+        fixExcludeAfter("\\d+".toRegex().matches(pathPart))
+    }
+
+    /** The function set a flag whether to create this button or not
+     *
+     * @param index
+     */
+    fun isCreateButton(index: Int) = when (circumstance) {
+        SituationUrlClassification.KEYWORD_PROCESS_LAST_SEGMENTS -> index >= sizeUrlSegments - 1 - processLastSegments
+        SituationUrlClassification.KEYWORD_NOT_PROCESS -> false
+        else -> true
+    }
+
+    /** The function
+     *
+     * @param href
+     */
+    private fun findExclude(href: String) {
+        circumstance = if (href.contains(OrganizationMenuBar.regexForUrlClassification) || href.contains(ProjectMenuBar.regexForUrlClassification)) {
+            SituationUrlClassification.PROJECT_OR_ORGANIZATION
+        } else if (href.contains(BenchmarkCategoryEnum.regexForUrlClassification)) {
+            SituationUrlClassification.ARCHIVE
+        } else if (href.contains(Regex("/[^/]+/[^/]+/history/execution/[1234567890]+/details"))) {
+            SituationUrlClassification.DETAILS
+        } else if (href.contains(Regex("/[^/]+/[^/]+/history/execution"))) {
+            SituationUrlClassification.EXECUTION
+        } else {
+            SituationUrlClassification.KEYWORD_PROCESS
+        }
+    }
+
+    /** Function check exclude and generate currentPath after the buttons creating
+     *
+     * @param pathPart
+     */
+    private fun fixCurrentPathAfter(pathPart: String) {
+        currentPath = when (circumstance) {
+            SituationUrlClassification.PROJECT_OR_ORGANIZATION, SituationUrlClassification.ARCHIVE -> "#"
+            SituationUrlClassification.DETAILS, SituationUrlClassification.EXECUTION -> if (pathPart == "execution") mergeUrls(pathPart) else currentPath
+            else -> currentPath
+        }
+    }
+
+    /** The function changes the exclude after the button is created
+     *
+     * @param isNumber
+     */
```

Comment:
```suggestion
    /**
     * The function changes the exclude after the button is created
     *
     * @param isNumber
     */
```

Reply:
Done

Reply:
btw, @akuleshov7 maybe add diktat rule for it? It looks strange
```
/** some text
*
**/
```